qiskit_nature/operators/second_quantization/fermionic_op.py

```diff
@@ -0,0 +1,368 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Fermionic-particle Operator."""
+
+from typing import Dict, List, Optional, Tuple, Union, cast
+
+import numpy as np
+from qiskit.opflow import PauliSumOp
+
+from qiskit_nature import QiskitNatureError
+
+from .particle_op import ParticleOp
+# TODO: Remove this after Terra#5617 is merged.
+
+
+class FermionicOp(ParticleOp):
+    r"""
+    N-mode Fermionic operator.
+
+    **Label**
+
+    Allowed characters for the label are `I`, `-`, `+`, `N`, and, `E`.
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Label
+          - Mathematical Representation
+          - Meaning
+        * - `I`
+          - :math:`I`
+          - Identity operator
+        * - `-`
+          - :math:`c`
+          - Annihilation operator
+        * - `+`
+          - :math:`c^\dagger`
+          - Creation operator
+        * - `N`
+          - :math:`n = c^\dagger c`
+          - Number operator
+        * - `E`
+          - :math:`I - n = c c^\dagger`
```

Comment:
```suggestion
        * - `-`
          - :math:`a`
          - Annihilation operator
        * - `+`
          - :math:`a^\dagger`
          - Creation operator
        * - `N`
          - :math:`N = a^\dagger a`
          - Number operator
        * - `E`
          - :math:`I - N = a a^\dagger`
```

Reply:
I would like to use `c` for fermion annihilation and `b` for boson. I would like to save `a` for an usual harmonic oscillator. This may be the notation of physicists. What do you think @pbark?

Reply:
Sure no problem :+1: If you have a clear plan for a consistent naming that is fine by me :slightly_smiling_face:

Reply:
Could we still make `n` and `e` capital letters, though?

Reply:
I feel that `N` is total number operator (N = \sum_i n_i). So I use small letter `n` here and I didn't introduce `e` for mathematical expression.